openapi_tester/schema_tester.py

```diff
@@ -267,31 +269,33 @@ def _validate_format(schema_section: dict, data: Any) -> Optional[str]:
             valid = isinstance(data, bytes)
         elif schema_format in ["date", "date-time"]:
             parser = parse_date if schema_format == "date" else parse_datetime
-            try:
-                result = parser(data)
-                valid = result is not None
-            except ValueError:
-                valid = False
+            result = parser(data)
+            valid = result is not None
         return None if valid else VALIDATE_FORMAT_ERROR.format(expected=schema_section["format"], received=str(data))
 
     @staticmethod
     def _validate_openapi_type(schema_section: dict, data: Any) -> Optional[str]:
         valid = True
         schema_type: Optional[str] = schema_section.get("type")
-        if not schema_type and "properties" in schema_section:
-            schema_type = "object"
         if not schema_type:
-            return None
+            if "properties" in schema_section:
+                schema_type = "object"
+            elif "items" in schema_section:
+                schema_type = "array"
+            else:
+                return None
         if schema_type in ["file", "string"]:
             valid = isinstance(data, (str, bytes))
         elif schema_type == "integer":
-            valid = isinstance(data, int)
+            valid = isinstance(data, int) and not isinstance(data, bool)
```

Comment:
why?

Reply:
Because the `bool` class in Python inherits from integer. We therefore need to be careful about how we validate OAS types. 

```python
>>> isinstance(True, int)
True
```

The spec very clearly states that *only* `True`/`False` values are valid `boolean` OAS types (not truthy values like `1` to represent `True`), and so I would imagine `True`/`False` also should not be considered valid `number` or `integer` types.

In other words, this is a difference between the native python types behavior and the spec, and we therefore need this in place to make sure we're validating the types correctly. Without this, we would validate `True` with an integer schema as correct.

https://swagger.io/docs/specification/data-models/data-types/#boolean

Reply:
but the solution for this is not to add the redundant checks here, its to change the order:

```
>>> isinstance(False, bool)
True
>>> isinstance(1, bool)
False
```

you see?

Reply:
put the bool test above the int test and thats that

Reply:
Hmm, well this isn't broken logic; in fact it seems like isolated, each of these lines are how we should validate the OAS type - and your proposition will mean any reordering will break our validation in the future.

```python
>>> isinstance(False, int) and not isinstance(False, bool)
False
>>> isinstance(1, int) and not isinstance(1, bool)
True
```

But sure, I can reorder them and remove the extra type check

Reply:
This actually won't work. Even if we move the block up, this isn't the filtering element; we filter by oas type:

```python
elif schema_type == "boolean":
    valid = isinstance(data, bool)
elif schema_type == "integer":
    valid = isinstance(data, int) and not isinstance(data, bool)
elif schema_type == "number":
    valid = isinstance(data, (int, float)) and not isinstance(data, bool)
```

Reply:
The boolean block will never hit to eliminate the `isinstance(data, bool)` scenario. The whole point here is that we should prevent an `"integer"` type from being validated successfully if it's a python boolean, because even though that's fine in python, it's not in the OAS spec.

Reply:
Ok